pkg/ingester/ingester.go

```diff
@@ -936,16 +1035,27 @@ func (i *Ingester) pushSamplesToAppender(userID string, timeseries []mimirpb.Pre
 		// and out-of-order support is not enabled.
 		// TODO(jesus.vazquez) If we had too many old samples we might want to
 		// extend the fast path to fail early.
-		if outOfOrderWindow <= 0 && minAppendTimeAvailable &&
-			len(ts.Samples) > 0 && len(ts.Exemplars) == 0 && allOutOfBounds(ts.Samples, minAppendTime) {
-			stats.failedSamplesCount += len(ts.Samples)
-			stats.sampleOutOfBoundsCount += len(ts.Samples)
-
+		if outOfOrderWindow <= 0 && minAppendTimeAvailable && len(ts.Exemplars) == 0 &&
+			(len(ts.Samples) > 0 || len(ts.Histograms) > 0) &&
+			(len(ts.Samples) == 0 || allOutOfBoundsFloats(ts.Samples, minAppendTime)) &&
+			(len(ts.Histograms) == 0 || (i.limits.AcceptNativeHistograms(userID) && allOutOfBoundsHistograms(ts.Histograms, minAppendTime))) {
+
+			stats.failedSamplesCount += len(ts.Samples) + len(ts.Histograms)
```

Comment:
_Same comment here. I don't think we should sum the two, but summing them when reading and appropriate._